packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/RelationInputDataManager.js

```diff
@@ -66,14 +66,8 @@ export const RelationInputDataManager = ({
       endpoint: endpoints.search,
       pageParams: {
         ...defaultParams,
-        // TODO: fix me cause this sucks
-        entityId: isCreatingEntry
-          ? undefined
-          : componentUid && !componentId
-          ? undefined
-          : !componentUid && !componentId
-          ? initialData.id
-          : componentId,
+        // eslint-disable-next-line no-nested-ternary
+        entityId: isCreatingEntry ? undefined : isComponentRelation ? componentId : initialData.id,
```

Comment:
Could you add a test for this? This is very difficult to understand and I'd like to make sure we do not brake this by accident in the future. If you want I can create a PR based on yours and add the test myself.

Reply:
Think you might be better off adding a test for this as you've done a lot of the testing work with axios around relations 😄